myia/infer/inferrers.py

```diff
@@ -82,22 +82,168 @@ def getattr_inferrer(node, args, unif):
     return res
 
 
+def _unary_op_inference(unary_op):
+    """Create a generic inference for builtin unary operator functions."""
+
+    def inf(node, args, unif):
+        """Generic inferrer for builtin unary operator functions."""
+        (a_node,) = args
+        a_type = yield Require(a_node)
+        a_interface = a_type.tracks.interface
+        if not hasattr(a_interface, unary_op):
+            raise TypeError(f"No {unary_op} method for type {a_interface}")
+        new_node = node.graph.apply(getattr(a_interface, unary_op), a_node)
+        yield Replace(new_node)
+        res = yield Require(new_node)
+        return res
+
+    return inference_function(inf)
+
+
+def _bin_op_inference(bin_op, bin_rop):
+    """Create a generic inference for builtin binary operator functions."""
+
+    def inf(node, args, unif):
+        """Generic inferrer for builtin binary operator functions."""
+        a_node, b_node = args
+        a_type = yield Require(a_node)
+        b_type = yield Require(b_node)
+
+        if isinstance(a_type, data.GenericBase) or isinstance(
+            b_type, data.GenericBase
+        ):
+            # If there is any generic in operands, just unify them.
+            return autils.unify(a_type, b_type)[0]
+
+        a_interface = a_type.tracks.interface
+        b_interface = b_type.tracks.interface
+        if hasattr(a_interface, bin_op):
+            new_node = node.graph.apply(
+                getattr(a_interface, bin_op), a_node, b_node
+            )
+        elif hasattr(b_interface, bin_rop):
```

Comment:
Just adding a TODO to make sure we don't forget to adjust this when we have the capability (not yet).

```suggestion
        # TODO: In order to support user-defined graphs for bin_op, the case where
        # it returns NotImplemented will have to be handled.
        elif hasattr(b_interface, bin_rop):
```

Reply:
Ok!